src/lib/annotations/doc/DocAnnotator.js

```diff
@@ -880,59 +850,36 @@ class DocAnnotator extends Annotator {
         let consumed = false;
         let activeThread = null;
 
-        // Destroy any pending highlights on click outside the highlight
-        const pendingThreads = this.getThreadsWithStates(PENDING_STATES);
-        pendingThreads.forEach((thread) => {
-            if (thread.type === TYPES.point) {
-                thread.destroy();
-            } else {
-                thread.cancelFirstComment();
-            }
-        });
-
-        // Only filter through highlight threads on the current page
         const page = annotatorUtil.getPageInfo(event.target).page;
-        const pageThreads = this.getHighlightThreadsOnPage(page);
-        pageThreads.forEach((thread) => {
-            // We use this to prevent a mousedown from activating two different
-            // highlights at the same time - this tracks whether a delegated
-            // mousedown activated some highlight, and then informs the other
-            // keydown handlers to not activate
-            const threadActive = thread.onClick(event, consumed);
-            if (threadActive) {
-                activeThread = thread;
-            }
+        const pageThreads = this.getThreadsOnPage(page);
+
+        Object.values(pageThreads).forEach((thread) => {
+            if (PENDING_STATES.indexOf(thread.state) >= 0) {
```

Comment:
Don't we have a method along the lines of `threadIsPending()`?

Reply:
Yup! Good catch!